src/services/features/index.ts

```diff
@@ -0,0 +1,173 @@
+
+export async function fetchFeatures({
+  apiKey,
+  ctx,
+  ssEvalEnabled = false,
+}: {
+  apiKey: string;
+  ctx: Context;
+  ssEvalEnabled?: boolean;
+}) {
+  const path = `/api/features/${apiKey}`;
+  if (!featuresCache) {
+    return;
+  }
+  if (!ctx.growthbookApiHost) {
+    throw new Error("missing required context for fetching features");
+  }
+  const url = ctx.growthbookApiHost + path;
+
+  // debounce requests
+  let promise = activeFetches[url];
+
+  if (!promise) {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const headers: any = {
+      "User-Agent": `GrowthBook Proxy`,
+    };
+    if (ssEvalEnabled) {
+      if (!ctx.secretApiKey) {
+        throw new Error("missing required context for fetching features");
+      }
+      headers["Authorization"] = `Bearer ${ctx.secretApiKey}`;
+    }
+    promise = got
+      .get(url, { headers })
+      .json()
+      .catch((e) => logger.error(e, "Refresh stale cache error"))
+      .finally(() => delete activeFetches[url]);
+
+    activeFetches[url] = promise;
+  }
+
+  const payload = await promise;
+
+  if (payload) {
+    logger.debug("cache STALE, refreshing cache...");
+    const oldEntry = await featuresCache.get(apiKey);
+    await featuresCache?.set(apiKey, payload);
+
+    if (ctx.enableEventStream && eventStreamManager) {
+      eventStreamManager?.publish({
+        apiKey,
+        event: "features",
+        payload,
+        oldPayload: oldEntry?.payload,
+        ssEvalEnabled,
+      });
+    }
+
+    return { payload, oldEntry };
+  } else {
+    logger.error("Unable to parse response");
+    return;
+  }
+}
+
+export function evaluateFeatures({
+  payload,
+  attributes,
+  ctx,
+}: {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  payload: any;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  attributes: Record<string, any>;
+  ctx?: Context;
+}) {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const evaluatedFeatures: Record<string, any> = {};
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const evaluatedExperiments: any[] = [];
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const trackExperiments: { experiment: any; result: any }[] = [];
+
+  const features = payload?.features;
+  const experiments = payload?.experiments;
+  const context: GBContext = {
+    attributes,
+  };
+  if (features) {
+    context.features = features;
+  }
+  if (experiments) {
+    context.experiments = experiments;
+  }
+
+  if (features || experiments) {
+    const gb = new GrowthBook(context);
+    if (ctx?.verboseDebugging) {
+      gb.debug = true;
+    }
+
+    const gbFeatures = gb.getFeatures();
+    for (const key in gbFeatures) {
+      const feature = gbFeatures[key];
+      const result = gb.evalFeature(key);
+      if (result.on) {
+        // reduced feature definition
+        evaluatedFeatures[key] = { defaultValue: true };
```

Comment:
Is this the correct minimal format for an evaluated feature flag?

Reply:
If `result.source === "experiment"`:
```ts
{
  defaultValue: result.value,
  rules: [
    {  
      force: result.value, 
      tracks: [
        {
          experiment: result.experiment,
          result: result
        }
     ]
   }
  ]
}
```

Otherwise:
```ts
{ defaultValue: result.value }
```

Reply:
make sure the SDK doesn't actually push the track payload for this case (deferred tracking)